content/news/2022-01-06-bevy-0.6/index.md

```diff
@@ -0,0 +1,1007 @@
+
+![bevymark](bevymark.png)
+
+
+### Sprite Ergonomics
+
+Sprite entities are now simpler to spawn:
+
+```rust
+// Old (Bevy 0.5)
+fn spawn_sprite(
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+    mut materials: ResMut<Assets<ColorMaterial>>,
+) {
+    let texture_handle = asset_server.load("player.png");
+    commands.spawn_bundle(SpriteBundle {
+        material: materials.add(texture_handle.into()),
+        ..Default::default()
+    });
+}
+
+// New (Bevy 0.6)
+fn spawn_sprite(mut commands: Commands, asset_server: Res<AssetServer>) {
+    commands.spawn_bundle(SpriteBundle {
+        texture: asset_server.load("player.png"),
+        ..Default::default()
+    });
+}
+```
+
+No need to manage sprite materials! Their texture handle is now a direct component and color can now be set directly on the {{rust_type(type="struct" crate="bevy_sprite" version="0.6.0" name="Sprite")}} component.
+
+### WGSL Shaders
+
+Bevy now uses [WGSL](https://www.w3.org/TR/WGSL/) for our built-in shaders and examples. WGSL is a new shader language being developed for WebGPU (although it is a "cross platform" shader language just like GLSL). Bevy still supports GLSL shaders, but WGSL is nice enough that, for now, we are treating it as our "officially recommended" shader language. WGSL is still being developed and polished, but given how much investment it is receiving I believe it is worth betting on. Consider this the start of the "official Bevy shader language" conversation, not the end of it.
+
+```rust
+[[group(0), binding(0)]]
+var<uniform> view: View;
+
+[[group(1), binding(0)]]
+var<uniform> mesh: Mesh;
+
+struct Vertex {
+    [[location(0)]] position: vec3<f32>;
+};
+
+struct VertexOutput {
+    [[builtin(position)]] clip_position: vec4<f32>;
+};
+
+[[stage(vertex)]]
+fn vertex(vertex: Vertex) -> VertexOutput {
+    var out: VertexOutput;
+    out.clip_position = view.view_proj * mesh.model * vec4<f32>(vertex.position, 1.0);
+    return out;
+}
+```
+
+
+### Shader Preprocessor
+
+<div class="release-feature-authors">authors: @cart, Rob Swain (@superdump), @mockersf</div>
+
+Bevy now has its own custom shader preprocessor. It currently supports `# import`, `# ifdef FOO`, `# ifndef FOO`, `# else`, and `# endif`, but we will expand it with more features as our needs grow.
+
+Shader preprocessors are often used to selectively enable shader code:
+
+```rust
+#ifdef TEXTURE
+[[group(1), binding(0)]]
+var sprite_texture: texture_2d<f32>;
+#endif
+```
+
+This pattern is very useful when defining complicated / configurable shaders (such as Bevy's PBR shader).
+
+### Shader Imports
+
+<div class="release-feature-authors">authors: @cart</div>
+
+The new preprocessor supports importing other shader files (which pulls in their entire contents). This comes in two forms:
+
+Asset path imports:
+
+```rust
+# import "shaders/cool_function.wgsl"
```

Comment:
in `pbr.wgsl` we use `#import` instead of `# import`, is this a deliberate style decision or a typo?

Reply:
I removed all these spaces in a suggestion. I don't like them. :)

Reply:
Yeah consistency is good and `#import` _is_ the style we landed on.